tests/behaviour/typeql/steps.rs

```diff
@@ -88,4 +109,85 @@ generic_step_impl! {
         assert!(result.is_err());
         assert!(result.unwrap_err().contains(&exception));
     }
+
+    #[step(expr = "typeql delete")]
+    async fn typeql_delete(context: &mut Context, step: &Step) {
+        let parsed = parse_query(step.docstring().unwrap());
+        assert!(parsed.is_ok());
+        let res = context.transaction().query().delete(&parsed.unwrap().to_string()).await;
+        assert!(res.is_ok());
+    }
+
+    #[step(expr = "typeql delete; throws exception")]
+    async fn typeql_delete_throws(context: &mut Context, step: &Step) {
+        let parsed = parse_query(step.docstring().unwrap());
+        if parsed.is_ok() {
+            let res = context.transaction().query().delete(&parsed.unwrap().to_string()).await;
+            assert!(res.is_err());
+        }
+    }
+
+    #[step(expr = "get answers of typeql match")]
+    async fn get_answers_of_typeql_match(context: &mut Context, step: &Step) {
+        let parsed = parse_query(step.docstring().unwrap());
+        assert!(parsed.is_ok());
+        let stream = context.transaction().query().match_(&parsed.unwrap().to_string());
+        assert!(stream.is_ok());
+        let res = stream.unwrap().try_collect::<Vec<_>>().await;
+        assert!(res.is_ok());
+        context.answer = res.unwrap();
+    }
+
+    #[step(expr = "answer size is: {int}")]
+    async fn answer_size(context: &mut Context, step: &Step, expected_answers: usize) {
+        let actual_answers = context.answer.len();
+        assert_eq!(
+            actual_answers, expected_answers,
+            "The number of identifier entries (rows) should match the number of answers, \
+            but found {expected_answers} identifier entries and {actual_answers} answers."
+        );
+    }
+
+    #[step(expr = "uniquely identify answer concepts")]
+    async fn uniquely_identify_answer_concepts(context: &mut Context, step: &Step) {
+        let step_table = iter_map_table(step).collect::<Vec<_>>();
+        let expected_answers = step_table.len();
+        let actual_answers = context.answer.len();
+        assert_eq!(
+            actual_answers, expected_answers,
+            "The number of identifier entries (rows) should match the number of answers, \
+            but found {expected_answers} identifier entries and {actual_answers} answers."
+        );
+        let mut matched_rows = 0;
+        for ans_row in &context.answer {
+            let ans_row_map: HashMap<&String, String> =
+                HashMap::from_iter(ans_row.map.iter().map(|(key, value)| (key, concept_as_string(value))));
+            for table_row in &step_table {
+                if ans_row_map.len() == table_row.len()
+                    && ans_row_map.iter().all(|(key, value)| table_row.get(key).map_or(false, |v| value == *v))
+                {
+                    matched_rows += 1;
+                    break;
+                }
+            }
```

Comment:
This is still very difficult to read and could benefit from introducing some helper functions.

Reply:
`answers_equal` function added.